membership-common/src/main/scala/com/gu/memsub/subsv2/Subscription.scala

```diff
@@ -84,17 +67,22 @@ object GetCurrentPlans {
        */
       val ensureStarted = unvalidated.ensure(DiscardedPlan(plan, s"hasn't started as of $dateToCheck").wrapNel)(_)
       val alreadyStarted =
-        if (plan.product == Contribution)
+        if (product == Contribution)
           ensureStarted(_ => sub.startDate <= date)
         else
           ensureStarted(_.start <= dateToCheck)
       val contributorPlanCancelled =
         alreadyStarted.ensure(DiscardedPlan(plan, "has a contributor plan which has been cancelled or removed").wrapNel)(_)
       val paidPlanEnded = alreadyStarted.ensure(DiscardedPlan(plan, "has a paid plan which has ended").wrapNel)(_)
       val digipackGiftEnded = alreadyStarted.ensure(DiscardedPlan(plan, "has a digipack gift plan which has ended").wrapNel)(_)
-      if (plan.product == Product.Contribution)
+      if (product == Product.Contribution)
         contributorPlanCancelled(_ => !sub.isCancelled && !plan.lastChangeType.contains("Remove"))
-      else if (plan.product == Product.Digipack && plan.charges.billingPeriod == OneTimeChargeBillingPeriod)
+      else if (
+        product == Product.Digipack && plan.billingPeriod
+          .leftMap(e => throw new RuntimeException("no billing period: " + e))
+          .toOption
+          .get == OneTimeChargeBillingPeriod
```

Comment:
now that billingPeriod is generated on the fly we have to deal with the error at this point rather than it failing when it's read in originally.